rfcs/0002-borrow-type.md

```diff
@@ -0,0 +1,257 @@
+- Feature Name: `borrow_type`
+- Start Date: 2022-05-27
+- RFC PR: [microsoft/windows-rs#1788](https://github.com/microsoft/windows-rs/pull/1788)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces a new type `Borrow<'a, T>` which mirrors the semantics of a `&'a T` but with the difference that it has the same in memory layout as `T`.
+
+# Motivation
+[motivation]: #motivation
+
+Windows APIs often have ownership semantics that differ slightly from Rust's model. Types can often have different ownership semantics depending on the context. In Rust we would normally model these different semantics by using two different types, in C/C++ this is usually left to documentation and convention. 
+
+This difference is only interesting for types that are *not* "blittable" (`Copy` in Rust parlance). Non-"blittable" types have non-trivial destructors and cannot just be arbitrarily mem-copied. "Blittable" types like integers and booleans on other the hand can be, and so their ownership semantics are simple and map directly to Rust's model.
```

Comment:
We should make a note about `PCWSTR` and kin. These types can be considerable "blittable" in this context since they have "trivial" ownership semantics from Rust's perspective (i.e., the user is in full control). This is the same reason `*mut T` and `*const T` are considered `Copy`. 

With this reasoning, I would see no reason for arguments of type `PCWSTR` to be modeled as `Borrow<'a, PCWSTR>` instead of just plain `PCWSTR` or *perhaps* `Into<PCWSTR>`.